src/core2/graph.test.js

```diff
@@ -577,7 +585,11 @@ describe("graph", () => {
     });
 
     it("throws for null or undefined address", () => {
-      const e = (address: any) => ({...edge(), address});
+      const e = (address: any) => {
+        const edge = new SimpleEdge(srcRef(), dstRef());
+        (edge: any).address = () => address;
```

Comment:
Eek. How about instead:

```js
const e = (address: null | void) =>
  new class extends SimpleEdge {
    address() {
      return (address: any);
    }
  }(srcRef(), dstRef());
```

As you use similar things a bunch below, maybe

```js
class ShiftyEdge extends SimpleEdge {
  _overrides: {|+src?: any, +dst?: any, +address?: any|};
  constructor(overrides, ...args) {
    super(...args);
    this._overrides = overrides;
  }
  src() {
    return "src" in this._overrides ? this._overrides.src : super.src();
  }
  dst() {
    return "dst" in this._overrides ? this._overrides.dst : super.dst();
  }
  address() {
    return "address" in this._overrides
      ? this._overrides.address
      : super.address();
  }
}
```

?